### PR TITLE
Nathan/style location category component

### PR DESCRIPTION
**Pull request by nathanhuynhers**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Implemented clickable <LocationCategory /> components that trigger the new LocationCategoryPanel
- Styled the new panel to match Figma, including:
- Back arrow
- Category title
- Exit (X) button
- Search bar UI (no functionality yet)
- Panel dynamically renders options based on category selection:

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

image
image
image
image
image

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

- Start by clicking the Location dropdown
- Select a category (e.g. County, Region, etc.)
- Confirm the corresponding panel opens with:
- Check that header navigation works
- See if scroll functionality works
- Check that search is an input box
- Test the up arrow – it should collapse the dropdown using the same logic as the main button

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."

Town is currently empty because there are 933 towns so we are going to see what to do there.

image
Want to also talk about if user should be able to select whole row or only click the arrow/button. Will have to discuss with our goated designer to get some thoughts. Right now I set the cursor to be a pointer (hand) if hovered over a row, but it seems a bit off. If anyone has any ideas please contribute!

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @jjstnlee

### CODE REVIEW

File: components/LocationCategoryPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  handleButtonClick: (filter: FilterType) => void;
+  currFilter: FilterType;
+}
+
+export default function LocationCategoryPanel(
+  props: LocationCategoryPanelProps,
+) {
+  const [selectedItem, setSelectedItem] = useState<string | null>(null);
+
+  const counties = [
```
Comment (ethan-tam33) [inline components/LocationCategoryPanel/index.tsx:43]: @jjstnlee did we confirm that we will get these values from the database instead? if so, make that change here

File: components/LocationCategoryPanel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    switch (category) {
+      case 'County':
+        return counties; // your 62-county array
+      case 'Region':
+        return nyRegions; // your region list
+      case 'Utility Service Territory':
+        return utilityServiceTerritories; // the list from earlier
+      case 'State Senate District':
+        return Array.from(
+          { length: 63 },
```
Comment (ethan-tam33) [inline components/LocationCategoryPanel/index.tsx:190]: let's not hardcode!

File: components/LocationCategoryPanel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+          (_, i) => `State Senate District ${i + 1}`,
+        );
+      case 'Assembly District':
+        return Array.from(
+          { length: 150 },
```
Comment (ethan-tam33) [inline components/LocationCategoryPanel/index.tsx:195]: let's not hardcode!

**Review by ethan-tam33 — CHANGES_REQUESTED** [review]
Looks great! In addition to the small things I commented:
- idk if this is too important but the scroll bar for "Region" panel seems slower than the scroll bar in the other panels?
- this might not be in-scope for this sprint, but when I click on some options and close the dropdown (with or without clicking on the Apply button), the options I selected should be shown as such when I reopen it. Look into how the other dropdowns are maintaining this state even when closed so you can fully focus on functionality later

**Comment by nathanhuynhers** [comment]
"this might not be in-scope for this sprint, but when I click on some options and close the dropdown (with or without clicking on the Apply button), the options I selected should be shown as such when I reopen it. Look into how the other dropdowns are maintaining this state even when closed so you can fully focus on functionality later"

fixed in my most current sprint, just waiting to push

File: components/SearchBar/index.tsx [hunk 4]
```diff
@@ -26,27 +25,25 @@ export const SearchBar = ({
...
+      <SearchBarWrapperDiv>
+        {searchTerm && searchTerm.length > 0 ? (
+          <SearchExitButton
+            onClick={handleExit}
+            $isZero={searchTerm?.length === 0}
+          >
+            <SearchExit />
+          </SearchExitButton>
+        ) : (
+          <SearchIcon fill={'#4C5671'} />
```
Comment (jjstnlee) [inline components/SearchBar/index.tsx:38]: use `COLORS.navy85`

File: styles/texts.ts [hunk 5]
```diff
@@ -275,6 +275,44 @@ export const LocationCategoryText = styled.p<TextProps>`
...
+  color: ${COLORS.electricBlue};
+`;
+
+export const SearchInput = styled.input<TextProps>`
+  ${TextStylesCoinbaseText}
+  border: none;
+  outline: none;
+  width: 100%;
+  background: transparent;
+  color: rgba(46, 58, 89, 0.5);
```
Comment (jjstnlee) [inline styles/texts.ts:304]: use `COLORS.navy50`

**Review by jjstnlee — CHANGES_REQUESTED** [review]
